src/sysusers/sysusers.c

```diff
@@ -163,6 +181,66 @@ static void maybe_emit_login_defs_warning(Context *c) {
         c->login_defs_need_warning = false;
 }
 
+static void log_audit_op(
+                Context *c,
+                int type,
+                const char *op,
+                const char *name,
+                bool success) {
+#if HAVE_AUDIT
+        assert(c);
+
+        if (arg_dry_run || c->audit_fd < 0)
+                return;
+
+        log_trace("Emitting audit message: %s, %s, %u, %s",
+                  op, strna(name), id, success ? "good" : "bad");
+
+        /* Note: we send id == -1, even though we know the number, in particular on success. This is because
+         * if we send the id, the generated audit message will not contain the name. The name seems more
+         * useful than the number, hence send just the name:
+         *
+         * type=ADD_USER msg=audit(01/10/2025 16:02:00.639:3854) :
+         *   pid=3846380 uid=root auid=zbyszek ses=2 msg='op=adding user id=unknown(952) exe=systemd-sysusers ... res=success'
+         * vs.
+         * type=ADD_USER msg=audit(01/10/2025 16:03:15.457:3908) :
+         *   pid=3846607 uid=root auid=zbyszek ses=2 msg='op=adding user acct=foo5 exe=systemd-sysusers ... res=success'
+         */
+
+        audit_log_acct_message(
+                        c->audit_fd,
+                        type,
+                        program_invocation_short_name,
+                        op, name,
+                        /* id= */ -1,
```

Comment:
so i guess libaudit should actually have used the little known `id_t` posix type here (which is actually a generic type for uid_t + gid_t)....

either way, I think this should be passed as `(unsigned) -1` to highlight the type confusion, and our way out of it.